### DISCUSSION TITLE
some doubts about comsumer in apisix

### DISCUSSION THREAD

**Discussion by li-zheng-hao** [apache/apisix#8370] [Q&A] [answered]
After reading documentation, I still have some doubts about comsumer：

1. Does consumer refer to a user (jack, marry) or a class of users (administrator, buyers, sellers)? From the documentation it seems to refer to users, does this mean I need to store all user data in apisix? (I'd like to store user data in the database, but that seems to mean I have to use the forward-auth plugin to do the authentication)

**Comment by tokers** [comment, marked as answer]
> After reading documentation, I still have some doubts about comsumer：
> 
> 
> 
> 1. Does consumer refer to a user (jack, marry) or a class of users (administrator, buyers, sellers)? From the documentation it seems to refer to users, does this mean I need to store all user data in apisix? (I'd like to store user data in the database, but that seems to mean I have to use the forward-auth plugin to do the authentication)
> 
> 

Consumer is the abstraction for the api consumer. You don't have to store any user specific data in it.

**Comment by tzssangglass** [comment]
> 1. does this mean I need to store all user data in apisix?

no, user data should be stored in IdP server or other authentication services

**Comment by li-zheng-hao** [comment]
@tzssangglass @tokers i got it ,thanks!
